llvm/lib/Target/AIE/AIELegalizerHelper.cpp

```diff
@@ -153,17 +162,20 @@ bool AIELegalizerHelper::legalizeG_BUILD_VECTOR(LegalizerHelper &Helper,
 
   const AIEBaseInstrInfo *II = ST.getInstrInfo();
   MachineOperand *OperandBegin = MI.operands_begin(),
-                 *Operand = MI.operands_end() - 1;
-  while (Operand != OperandBegin) {
+                 *Operand = MI.operands_begin() + 1,
+                 *OperandEnd = MI.operands_end();
+  while (Operand != OperandEnd) {
```

Comment:
Any particular reason why we are reversing the iteration order?

Reply:
As we changed corresponding instruction of `G_AIE_ADD_VECTOR_ELT_LEFT` from vpush.lo to vpush.hi.

Reply:
Oh so actually it's the previous order that was "un-natural", now it all makes sense.